libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -448,6 +448,32 @@ void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uin
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
                           rx_joy_topic.axes[0], rx_joy_topic.axes[1], rx_joy_topic.axes[2], rx_joy_topic.axes[3]);
             // TODO implement joystick RC control to AP
+                    const uint32_t tnow = AP_HAL::millis();
+
+            //! @todo - to use same convention as MAVLink spec?
+            // see: void GCS_MAVLINK::handle_rc_channels_override
+
+            for (uint8_t i=0; i<MIN(8U, rx_joy_topic.axes_size); i++) {
+            const uint16_t override_data = static_cast<uint16_t>(rx_joy_topic.axes[i]);
+
+            // Per MAVLink spec a value of UINT16_MAX means to ignore this field.
+            if (override_data != UINT16_MAX) {
+                const uint16_t mapped_data=mapValue(override_data,-1,1,1000,2000);
```

Comment:
Use `linear_interpolate` from `AP_Math.h`. The upper and lower input values should be floats (so `-1.0` or `-1.0f`). The upper and lower output values should be obtained from the `RCx_MIN`and `RC_MAX` parameters (x=1, 2, 3, 4, ...).